ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -145,11 +145,19 @@ getCompletionsLSP lsp ide
               (Just pfix', _) -> do
                 let clientCaps = clientCapabilities $ shakeExtras ide
                 snippets <- WithSnippets . completionSnippetsOn <$> getClientConfig lsp
-                Completions . List <$> getCompletions ideOpts cci' parsedMod bindMap pfix' clientCaps snippets
+                allCompletions <- getCompletions ideOpts cci' parsedMod bindMap pfix' clientCaps snippets
+                let topCompletions = List $ take 20 allCompletions
```

Comment:
`let (topCompletions, restCompletions) = splitAt 20 allCompletions` and then use `null restCompletions` to figure out whether we are complete.

Reply:
Also, could the 20 be pulled out as a constant and commented about why we restrict things, and why 20 (or 100 or whatever) seems plausible.